Jenkinsfile

```diff
@@ -324,16 +324,13 @@ void publishMaven(mvnArgs='') {
         # Configure the client
         setup-garasign-client
 
+        cat $HOME/.config/Garantir/GRS/config.ini
+        sudo /opt/Garantir/bin/grsgpgconfig.sh
+
         # Load GPG key from the server
         GrsGPGLoader
 
-        # Place config in an expected location
-        # warning: don't change EOF indentation!
-        awk '$1=$1' << EOF > /home/jenkins/garasignconfig.txt
-          name = GaraSign
-          library = /usr/local/lib/Garantir/GRS/libgrsp11.so
-          slotListIndex = 0
-EOF
+        export SIGNING_KEYID=$(grep 'Key ID' $HOME/.gnupggrs/keysinfo.txt | awk 'NR==1{print $5}')
```

Comment:
Is there any value in having two `sh` steps for this? Why don't we just move the `mvn deploy` command into the same `sh` step block and we don't need to export the env var.

Reply:
We export it for pom's `keyname`. Two `sh` probably not needed, depends on groovy interpretation of `'''`

Reply:
yep, I did it for interpretation, maven fails with "repository element was not specified in the POM inside distributionManagement element". Guess I can switch to `"""` for multiline block

Reply:
fair point, I've just been bitten many times in the past trying to get environment variables happily from one `sh` step to another when they aren't defined by Jenkins itself, but the export is clearly working in this case (at least for now :smile:)

If we run into trouble in future we could always move this subshell grep into the maven deploy sh step since the keysinfo can be read at any point  - that would avoid most of the escaping.